src/App/VPSScan/Scan.hs

```diff
@@ -93,4 +106,16 @@ runIPRScan basedir scanId vpsOpts@VPSOpts{..} =
       trace "[IPR] Post to Scotland Yard complete"
       trace "[IPR] IPR scan complete"
     Nothing ->
-      trace "[IPR] IPR Scan disabled"
\ No newline at end of file
+      trace "[IPR] IPR Scan disabled"
+
+runS3Upload ::
+  ( Has Diagnostics sig m
+  , Has Trace sig m
+  , MonadIO m
+  ) => Path Abs Dir -> Text -> VPSOpts -> String -> m ()
+runS3Upload basedir scanId VPSOpts{..} bucket =
```

Comment:
[nit] can we pass IPROpts in here directly? It seems that the caller already does the case split